uniffi/src/scaffolding.rs

```diff
@@ -32,8 +32,15 @@ mod filters {
             TypeReference::String => "&str".to_string(),
             TypeReference::Enum(name) => name.clone(),
             TypeReference::Record(name) => name.clone(),
-            TypeReference::Optional(t) => format!("Option<{}>", type_rs(t)?),
-            TypeReference::Sequence(t) => format!("Vec<{}>", type_rs(t)?),
+
+            // TODO: Clean this up, the `ret_type_rs` is supposed to only be used on return types
+            // and solve the issue of asymmetry with strings.
+            // When a string is a record member or a part of a sequence, we use the `String` type even if the string is an argument
+            // However, when we use strings directly, we use an &str as an argument.
+            // This reeks of future issues, a possible solution is to abandon the asymmetry (and FfiStr) and just use
+            // `String` everywhere as a ByteBuffer and give up an extra memory allocation
```

Comment:
This makes sense in the short term. I've been noodling a bit on the problems we've had around the asymmetry with strings here, and e.g. the `RawPointer` hack we had to add for the destructor. I think there's a bit of cleanup we can do in the typing here to simplify it, but I don't fully have my head around it yet. This is a great extra input to consider.

Reply:
Could you please file a follow-up issue for this?

Reply:
#182 ;)